core/arch/arm/kernel/kern.ld.S

```diff
@@ -250,6 +268,10 @@ SECTIONS
 		__text_init_end = .;
 	}
 
+	. = ALIGN(SMALL_PAGE_SIZE);
```

Comment:
Is it that harmful to mix ro-data with code in a page?
If an attacker is able to select some address to jump to the code pages has to be so much have so much more potential to contain useful code any way.